### PR TITLE
Support advance specified in write()

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**

### CODE REVIEW

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -1400,6 +1400,35 @@ int compile_to_object_file_c(const std::string &infile,
     return 0;
 }
 
+int runCommandAndGetStdout(const std::string& command, std::string &stdoutOutput) {
```
Comment (certik) [inline src/bin/lfortran.cpp:1403]: To be consistent with the rest of the file:
```suggestion
int run_command_and_get_stdout(const std::string& command, std::string &stdoutOutput) {
```

File: src/bin/lfortran.cpp [hunk 2]
```diff
@@ -1522,29 +1538,35 @@ int link_executable(const std::vector<std::string> &infiles,
...
+        }
+        int err = system(compile_cmd.c_str());
+        if (err) {
+            std::cout << "The command '" + compile_cmd + "' failed." << std::endl;
+            return 10;
+        }
+        if (compiler_options.exec_output) {
+            std::string exec_stdout;
+            int err = runCommandAndGetStdout(run_cmd, exec_stdout);
+            std::cout << exec_stdout;
```
Comment (certik) [inline src/bin/lfortran.cpp:1561]: What is the point of capturing and then printing it?

Why cannot you just let it print and capture it in Python?

Comment (Shaikh-Ubaid) [reply src/bin/lfortran.cpp]: >Why cannot you just let it print and capture it in Python?

That is a good idea. Thank you! I updated it.

File: src/libasr/pass/print_arr.cpp [hunk 3]
```diff
@@ -256,6 +256,18 @@ class PrintArrVisitor : public PassUtils::PassVisitor<PrintArrVisitor>
         return doloop;
     }
 
+    void print_args_apart_from_arrays(std::vector<ASR::expr_t*> &write_body, const ASR::FileWrite_t& x) {
+        Vec<ASR::expr_t*> body;
+        body.reserve(al, write_body.size());
+        for (size_t j=0; j<write_body.size(); j++) {
+            body.push_back(al, write_body[j]);
+        }
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/print_arr.cpp:264]:
```suggestion
        Vec<ASR::expr_t*> body;
        body.from_pointer_n(&write_body[0], write_body.size())
```
I forgot the function name, try `from_pointer_n` or `from_pointer_n_copy`

**Review by Thirumalai-Shaktivel — APPROVED** [review]
LGTM, but let's get this reviewed by @certik or @czgdp1807 as well.
Thank you!

File: run_tests.py [hunk 4]
```diff
@@ -383,6 +384,14 @@ def is_included(backend):
         run_test(filename, "bin", "lfortran --no-color {infile} -o {outfile}",
                  filename, update_reference, extra_args)
 
+    if execute:
+        if no_llvm:
+            log.info(f"{filename} * obj    SKIPPED as requested")
+        else:
+            run_test(filename, "execute", "lfortran --no-color {infile}",
```
Comment (Thirumalai-Shaktivel) [inline run_tests.py:391]: We have the same option in LPython, which says `run` or `runtime`. Do you want to keep them the same name?
https://github.com/lcompilers/lpython/blob/3cf81f36531d6c880368e790099ac46bd2012ef1/run_tests.py#L140C1-L143C1

**Comment by Shaikh-Ubaid** [comment]
I noticed an issue in `./run_tests.py` (from the root directory). There is a race condition between executions of `tests/write4.f90` and `tests/write5.f90`. I think `./run_tests.py` runs tests in parallel and both these tests generate the `a.out` binary. In some cases one of the binary overwrites the other leading to unexpected results or failure of `./run_tests.py`. I think we need unique names for the generated binaries. We currently only execute the binary if it is named `a.out`. Thus, I think we probably also need a flag in lfortran `--execute-bin`, so that when this flag is provided, we execute the generated binary (of custom name).

File: run_tests.py (hunk 4 above)
Comment (Shaikh-Ubaid) [reply run_tests.py:391]: We can keep them same names to avoid confusion.

**Comment by Shaikh-Ubaid** [comment]
>I noticed an issue in ./run_tests.py (from the root directory). There is a race condition between executions of tests/write4.f90 and tests/write5.f90

It seems this issue does not occur in LPython, since it names the generated binaries as per the input file name, thus generating unique binaries.

**Comment by Thirumalai-Shaktivel** [comment]
Yup, I have seen this in the LPython (For which, I came up with a unique name for each output file).
You can also save the runtime output in the sequential execution instead of parallel. In the same way, as we do for the `extrafiles` option, see: https://github.com/lfortran/lfortran/pull/1231/files.

**Comment by Shaikh-Ubaid** [comment]
>You can also save the runtime output in the sequential execution instead of parallel. In the same way, as we do for the extrafiles option, see: https://github.com/lfortran/lfortran/pull/1231/files.

I think serial execution would get slow as more tests are added. So, I would use a parallel approach if possible/available. In our case it is possible by adding a new flag or naming according to the input file. So, I think I would use that.

**Comment by Shaikh-Ubaid** [comment]
It seems compilers traditionally name the generated binary as `a.out`. Since Fortran is a compiled langauge, I guess we should follow the same `a.out` binary name by default. Therefore, I think we should add a new flag `--execute-bin` or any other name.

**Review by certik — APPROVED** [review]
I think it's fine, thanks!

File: run_tests.py [hunk 5]
```diff
@@ -383,6 +384,14 @@ def is_included(backend):
         run_test(filename, "bin", "lfortran --no-color {infile} -o {outfile}",
                  filename, update_reference, extra_args)
 
+    if run:
+        if no_llvm:
+            log.info(f"{filename} * obj    SKIPPED as requested")
+        else:
+            run_test(filename, "run", "lfortran --no-color --run {infile} -o {outfile}",
```
Comment (Thirumalai-Shaktivel) [inline run_tests.py:391]:
```suggestion
            run_test(filename, "runtime", "lfortran --no-color --run {infile} -o {outfile}",
```

**Pull request merged**